catalogue_api/api/src/main/scala/uk/ac/wellcome/platform/api/services/WorksService.scala

```diff
@@ -21,32 +24,31 @@ class WorksService @Inject()(searchService: ElasticsearchService)(
   implicit ec: ExecutionContext) {
 
   def findWorkById(canonicalId: String)(
-    documentOptions: ElasticsearchDocumentOptions)
-    : Future[Option[IdentifiedBaseWork]] =
+    index: Index): Future[Either[ElasticError, Option[IdentifiedBaseWork]]] =
     searchService
-      .findResultById(canonicalId)(documentOptions)
+      .findResultById(canonicalId)(index)
       .map { result =>
-        if (result.exists)
-          Some(jsonTo[IdentifiedBaseWork](result.sourceAsString))
-        else None
+        result.right.map { response: GetResponse =>
+          if (response.exists)
+            Some(jsonTo[IdentifiedBaseWork](response.sourceAsString))
+          else None
+        }
       }
 
-  def listWorks(documentOptions: ElasticsearchDocumentOptions,
-                worksSearchOptions: WorksSearchOptions): Future[ResultList] =
+  def listWorks(index: Index, worksSearchOptions: WorksSearchOptions)
+    : Future[Either[ElasticError, ResultList]] =
     searchService
-      .listResults(
-        documentOptions,
-        toElasticsearchQueryOptions(worksSearchOptions))
-      .map { createResultList }
+      .listResults(index, toElasticsearchQueryOptions(worksSearchOptions))
+      .map { _.right.map { createResultList } }
 
-  def searchWorks(query: String)(
-    documentOptions: ElasticsearchDocumentOptions,
-    worksSearchOptions: WorksSearchOptions): Future[ResultList] =
+  def searchWorks(query: String)(index: Index,
+                                 worksSearchOptions: WorksSearchOptions)
+    : Future[Either[ElasticError, ResultList]] =
     searchService
       .simpleStringQueryResults(query)(
-        documentOptions,
+        index,
         toElasticsearchQueryOptions(worksSearchOptions))
-      .map { createResultList }
+      .map { _.right.map { createResultList } }
```

Comment:
What happens if the either is a left? I think in general we should avoid getting the right projection in production code...It's a bit like doing .get on an Option

Reply:
Actually I think this is exactly the same as doing `_.map { createResultList }`, so no need to get the right projection

Reply:
KK. It’s not a `right.get` because I was trying to map only on the Right value. :-)

Reply:
Yeah, I was expecting that to fail with a left but it doesn't. It's still unnecessary in scala 2.12 because Eithers are right biased, so I think you can loose it